netmanager/src/views/components/AirqualitySearch/index.js

```diff
@@ -46,12 +46,25 @@ const AQSearch = () => {
     return null;
   };
 
+  const createAutocompleteRequest = (inputValue) => {
+    if (inputValue.length > 4) {
+      return {
+        input: inputValue
+      };
+    } else {
+      return {
+        input: ''
+      };
+    }
+  };
+
   return (
     <div>
       <MUIPlacesAutocomplete
         onSuggestionSelected={onSuggestionSelected}
         renderTarget={() => <div />}
         textFieldProps={{ variant: 'outlined', fullWidth: true, type: 'search', autoFocus: true }}
+        createAutocompleteRequest={createAutocompleteRequest}
```

Comment:
Hi @Codebmk , thanks for this. Much appreciated!

I am still not sure if this implementation achieves the desired outcome of running the **autocomplete** functionality only when the entered characters in the search box are more than 4. Here's why:

The `createAutocompleteRequest` function indeed checks the length of the` inputValue` and returns an empty string ('') if the length is not greater than 4. However, the `createAutocompleteRequest `function is used as a callback for `MUIPlacesAutocomplete`, which means it determines the content of the autocomplete suggestions based on the` inputValue` returned by this function. It doesn't prevent the autocomplete functionality from running when the characters are less than 4; it only affects the suggestions.

In other words, the autocomplete component itself will still be active and respond to user input even if the input length is less than 4 characters. It's just that, in such cases, the autocomplete suggestions will be empty because `createAutocompleteRequest `returns an empty string as the` input.`

If you want to completely disable the autocomplete functionality until there are at least 4 characters entered, you should conditionally render the `MUIPlacesAutocomplete` component based on the length of the input value. Here's an example of how you can achieve that:

```
return (
  inputValue.length > 4 ? (
    <MUIPlacesAutocomplete
      onSuggestionSelected={onSuggestionSelected}
      renderTarget={() => <div />}
      textFieldProps={{ variant: 'outlined', fullWidth: true, type: 'search', autoFocus: true }}
    />
  ) : (
    <input
      type="text"
      value={inputValue}
      onChange={(e) => setInputValue(e.target.value)}
      placeholder="Enter at least 5 characters to search"
    />
  )
);

```
In this code above, the `MUIPlacesAutocomplete` component is conditionally rendered only when the `inputValue` has more than 4 characters. Otherwise, a regular input field is shown with a placeholder instructing the user to enter at least 5 characters to search. This way, you prevent the autocomplete functionality from running until the condition is met.

Thanks again! Looking forward to your thoughts about the notes above.
cc: @BenjaminSsempala